src/vivarium_csu_alzheimers/components/testing.py

```diff
@@ -72,22 +74,35 @@ def on_initialize_simulants(self, pop_data: SimulantData) -> None:
             pop_data.index, additional_key=COLUMNS.TESTING_PROPENSITY
         )
         pop[COLUMNS.TESTING_STATE] = TESTING_STATES.NOT_TESTED
-        pop[COLUMNS.BBBM_TEST_DATE] = pd.NaT
         pop[COLUMNS.BBBM_TEST_RESULT] = BBBM_TEST_RESULTS.NOT_TESTED
         pop[COLUMNS.BBBM_TEST_EVER_ELIGIBLE] = False
+        pop[COLUMNS.BBBM_TEST_DATE] = pd.NaT
 
+        # Update to reflect history
+        event_time = pop_data.creation_time + get_timedelta_from_step_size(self.step_size)
+        bbbm_eligible_mask = self._get_bbbm_eligible_simulants(pop, event_time)
+        pop[COLUMNS.BBBM_TEST_DATE] = self._generate_bbbm_testing_history(
```

Comment:
It's probably fine for this rush job, but I don't like the sudden pattern change where here we return the column but then down below we update in-place.

This also sort of freaks me out b/c it looks here like you're going to update the  _entire_ columns but you're really not since it depends on the mask being passed in. 🤷